CMakeLists.txt

```diff
@@ -511,18 +511,27 @@ if(NOT AVIF_CODEC_AOM_ENABLED
     message(WARNING "libavif: No decoding library is enabled.")
 endif()
 
-set_target_properties(avif PROPERTIES VERSION ${LIBRARY_VERSION} SOVERSION ${LIBRARY_SOVERSION} C_VISIBILITY_PRESET hidden)
-target_include_directories(avif PUBLIC $<BUILD_INTERFACE:${libavif_SOURCE_DIR}/include> $<INSTALL_INTERFACE:include>)
+set_target_properties(avif_obj PROPERTIES C_VISIBILITY_PRESET hidden)
+set_property(TARGET avif_obj PROPERTY POSITION_INDEPENDENT_CODE ON)
+target_include_directories(avif_obj PUBLIC $<BUILD_INTERFACE:${libavif_SOURCE_DIR}/include>)
 if(NOT AVIF_LIBYUV_ENABLED)
-    target_include_directories(avif PRIVATE ${libavif_SOURCE_DIR}/third_party/libyuv/include/)
+    target_include_directories(avif_obj PRIVATE ${libavif_SOURCE_DIR}/third_party/libyuv/include/)
 endif()
 if(AVIF_ENABLE_COMPLIANCE_WARDEN)
-    target_include_directories(avif PRIVATE ${libavif_SOURCE_DIR}/ext/ComplianceWarden/src/utils/)
+    target_include_directories(avif_obj PRIVATE ${libavif_SOURCE_DIR}/ext/ComplianceWarden/src/utils/)
 endif()
 set(AVIF_PKG_CONFIG_EXTRA_CFLAGS "")
 if(BUILD_SHARED_LIBS)
-    target_compile_definitions(avif PUBLIC AVIF_DLL PRIVATE AVIF_BUILDING_SHARED_LIBS)
+    target_compile_definitions(avif_obj PUBLIC AVIF_DLL PRIVATE AVIF_BUILDING_SHARED_LIBS)
```

Comment:
Nit/optional: When building a file in src/, the compiler command line is:

> /usr/bin/cc -DAVIF_BUILDING_SHARED_LIBS -DAVIF_CODEC_AOM=1 -DAVIF_CODEC_AOM_DECODE=1 -DAVIF_CODEC_AOM_ENCODE=1 -DAVIF_DLL  ...

It would be good if `-DAVIF_BUILDING_SHARED_LIBS` and `-DAVIF_DLL` could be next to each other. I wonder if the current gap between them is caused by line 547.

Reply:
From what I see, definitions are sorted in alphabetical order. This is what we have right now on main:
```
/usr/bin/cc -DAVIF_BUILDING_SHARED_LIBS -DAVIF_CODEC_AOM=1 -DAVIF_CODEC_AOM_DECODE=1 -DAVIF_CODEC_AOM_ENCODE=1 -DAVIF_DLL -Davif_EXPORTS
```